framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -942,10 +942,18 @@ public AnnotatedTypeMirror getAnnotatedType(Tree tree) {
             return treeCache.get(tree).deepCopy();
         }
 
-        AnnotatedTypeMirror type;
         if (TreeUtils.isClassTree(tree)) {
-            type = fromClass((ClassTree)tree);
-        } else if (tree.getKind() == Tree.Kind.METHOD ||
+            ClassTree classTree = (ClassTree) tree;
+            AnnotatedTypeMirror type = getAnnotatedTypeOfClassTree(classTree);
```

Comment:
Yes, because that getAnnotatedTypeOfClassTree is called in other locations where the cache is not checked before the call.